### ISSUE TITLE
Uncaught TypeError: Cannot read property 'dispose' of undefined

### ISSUE THREAD

**Issue by valichek** [aurelia/i18n#37]
After updating to beta-1 I got in Chrome
```
Uncaught TypeError: Cannot read property 'dispose' of undefined
unbind @ t.js:102
unbind @ aurelia-templating.js:2767
unbind @ aurelia-templating.js:883
unbind @ if.js:68
unbind @ aurelia-templating.js:2767
unbind @ aurelia-templating.js:883
unbind @ aurelia-templating.js:2763
unbind @ aurelia-templating.js:883
(anonymous function) @ if.js:40
flushMicroTaskQueue @ aurelia-task-queue.js:110
(anonymous function) @ aurelia-task-queue.js:60
```

In Firefox I have `TypeError: this.subscription is undefined`

The error occurs when changing `merchantState.selector`
```html
// somewhere in app-subcomponent.html
<span if.bind="merchantState.selector != 'single'" data-toggle="dropdown" href="#" class="dropdown-toggle" style="cursor: pointer;">
     <span if.bind="merchantState.merchant">${merchantState.merchant.user.companyname</span>
     <span show.bind="!merchantState.merchant" t="merchant-selector.undefined"></span>
</span>
```
I change `merchantState.selector` during navigation in auth step, and app-subcomponent is removed at that moment with if.bind="..." from the DOM.

Just got some idea to override the problem. That html works with no  error.
```html
<span if.bind="merchantState.selector != 'single'" data-toggle="dropdown" href="#" class="dropdown-toggle" style="cursor: pointer;">
     <span if.bind="merchantState.merchant">${merchantState.merchant.user.companyname}</span>
     <template show.bind="!merchantState.merchant">
         <span t="merchant-selector.undefined"></span>
     </template>
</span>
```

@zewa666 tried
```html
<span t="merchant-selector.undefined" show.bind="!merchantState.merchant"></span>
```
but error is still there, and the error trace is the same

**Comment by zewa666** [comment]
ok I can't actually replicate the issue but have talked to @jdanyow to update [aurelia-plunker](http://plnkr.co/edit/5vMoxM?p=preview) to include it so you can easily create a sample. Stay tuned will reply here once its updated

**Comment by jdanyow** [comment]
@valichek / @zewa666 sorry for the delay- the plunker has been updated with the latest aurelia bits and now includes the aurelia-i18n plugin.

http://bit.ly/aurelia-plunker

**Comment by sternert** [comment]
I have the same issue, I have trouble to pinpoint where the actual problem is and are thus unable to show it in plunkr.
    Unhandled promise rejection TypeError: Cannot read property 'dispose' of undefined
        at TCustomAttribute.unbind (http://localhost:1111/jspm_packages/github/aurelia/i18n@0.2.6/t.js?bust=1448609338856:118:26)
        at Controller.unbind (http://localhost:1111/jspm_packages/github/aurelia/templating@0.16.0/aurelia-templating.js?bust=1448609338856:2724:20)
        at View.unbind (http://localhost:1111/jspm_packages/github/aurelia/templating@0.16.0/aurelia-templating.js?bust=1448609338856:841:26)
        at ViewSlot.unbind (http://localhost:1111/jspm_packages/github/aurelia/templating@0.16.0/aurelia-templating.js?bust=1448609338856:1137:21)
        at View.unbind (http://localhost:1111/jspm_packages/github/aurelia/templating@0.16.0/aurelia-templating.js?bust=1448609338856:846:23)
        at ViewFactory.returnViewToCache (http://localhost:1111/jspm_packages/github/aurelia/templating@0.16.0/aurelia-templating.js?bust=1448609338856:1731:14)
        at View.returnToCache (http://localhost:1111/jspm_packages/github/aurelia/templating@0.16.0/aurelia-templating.js?bust=1448609338856:753:24)
        at removeAction (http://localhost:1111/jspm_packages/github/aurelia/templating@0.16.0/aurelia-templating.js?bust=1448609338856:1246:25)
        at ViewSlot.removeAll (http://localhost:1111/jspm_packages/github/aurelia/templating@0.16.0/aurelia-templating.js?bust=1448609338856:1259:7)
        at http://localhost:1111/jspm_packages/github/aurelia/templating@0.16.0/aurelia-templating.js?bust=1448609338856:3653:53(anonymous function) @ :1111/jspm_packages/npm/core-js@1.2.5/modules/es6.promise.js?bust=1448609338856:122
    :1111/jspm_packages/npm/core-js@1.2.5/modules/es6.promise.js?bust=1448609338856:122 Unhandled promise rejection TypeError: Cannot read property 'dispose' of undefined(…)(anonymous function) @ :1111/jspm_packages/npm/core-js@1.2.5/modules/es6.promise.js?bust=1448609338856:122

**Comment by zewa666** [comment]
hey we currently have weird issues with the show.binding if used together with the animator-css. May I ask you guys to test if the error still happens if you remove the animator-css plugin?

**Comment by valichek** [comment]
I have `aurelia-animator-css` plugin disabled

**Comment by zewa666** [comment]
sry I seem to be doing something wrong, still can't replicate the issue. @valichek may I ask you to use the updated aurelia plunker to create an example depicting the issue?

**Comment by valichek** [comment]
@zewa666 will try

**Comment by jdanyow** [comment]
I wonder if there's a scenario where `bind` then `unbind` is called in the same task queue flush. If that were to happen, `unbind` would be called before [this line](https://github.com/aurelia/i18n/blob/master/src/t.js#L57) is evaluated.

Maybe this could be reproduced by putting a `t` custom attribute on a element that has `if.bind="show"` and then executing `this.show = !this.show` a few times in the view-model.

It might be good to clear these timeouts in `unbind`: 
* https://github.com/aurelia/i18n/blob/master/src/t.js#L49
* https://github.com/aurelia/i18n/blob/master/src/t.js#L61

Otherwise there is the potential for the deferred binding code to execute after unbind is called.

**Comment by Mordred** [comment]
Any progress with this? You should clear these timeouts during `unbind` and run `dispose()` of subscription only if really subscribed.

**Comment by zewa666** [comment]
@EisenbergEffect could you create a new release and @valichek check it afterwards? I guess @Mordred's PR should fix that

**Comment by EisenbergEffect** [comment]
We will have some new releases this week. Hopefully tomorrow or Wednesday. I'm waiting on a couple of final things to come in so we can aggregate the patches and release details.

**Comment by zewa666** [comment]
@valichek has this been fixed with the recent release?

**Comment by valichek** [comment]
yes, error is gone

**Comment by zewa666** [comment]
Allrighty :+1:
